src/tests/ftest/osa/osa_offline_drain.py

```diff
@@ -61,20 +65,27 @@ def run_offline_drain_test(self, num_pool, data=False):
             pool[val].nvme_size.value = int(pool[val].nvme_size.value /
                                             num_pool)
             pool[val].create()
-            pool_uuid.append(pool[val].uuid)
             self.pool = pool[val]
+            if drain_during_aggregation is True:
+                test_seq = self.ior_test_sequence[1]
+                self.pool.set_property("reclaim", "disabled")
+            else:
+                test_seq = self.ior_test_sequence[0]
+
             if data:
-                self.run_ior_thread("Write", self.ior_dfs_oclass[0],
-                                    self.ior_apis[0],
-                                    self.ior_test_sequence[0])
+                self.run_ior_thread("Write", oclass, test_seq)
+                self.run_mdtest_thread()
 
-        # Drain the pool_uuid, rank and targets
+        # Drain rank and targets
         for val in range(0, num_pool):
             self.pool = pool[val]
             rank = rank + val
             self.pool.display_pool_daos_space("Pool space: Beginning")
             pver_begin = self.get_pool_version()
             self.log.info("Pool Version at the beginning %s", pver_begin)
+            if drain_during_aggregation is True:
+                self.pool.set_property("reclaim", "time")
+                time.sleep(90)
```

Comment:
Just curious, why sleep(90) here?  Then aggregation might already finish its job before drain?

Reply:
As discussed, let me constantly, loop the exclude, reintegration operations for 180 seconds and see whether aggregation kicks in... Lot of aggregation scripts have this time delay for the aggregation to kick in after an IO operation is performed.

Reply:
If aggregation scripts have sleep in them, i think it's not a good approach. There is no concrete way we can determine with a sleep if aggregation started or finished.
I would suggest to use pool query results for verify aggregation if the data has been deleted after aggregation was enabled.

Reply:
Removed sleep and added the new method. Write ior data and delete the container to make sure aggregation happens.